Add 3.0 section for What's new (on the site) (#5116)
dart-lang/site-www: src/_guides/whats-new.md

[dart-announce]: https://groups.google.com/a/dartlang.org/d/forum/announce
[Dart blog]: https://medium.com/dartlang

## May 10, 2023: 3.0 release

This section lists notable changes made from January 26, 2023,
through May 10, 2023.
For details about the major 3.0 release,
check out [Announcing Dart 3][],
and the [SDK changelog][].

[Announcing Dart 3]: https://medium.com/dartlang/announcing-dart-3-53f065a10635
[SDK changelog]: https://github.com/dart-lang/sdk/blob/main/CHANGELOG.md#300---2023-05-10

### Docs updated or added to dart.dev
{: .no_toc}

In preparation for Dart 3, we split up and reorganized the

> constantly growing language tour into new conceptual categories.

Reviewer: ```suggestion
Language Tour into new conceptual categories.
```

You can access the restructured language documentation
under **Language** in the side navigation menu, or by
visiting the updated [Introduction to Dart][].

Taking advantage of this new structure,
we added documentation for each of Dart 3's major features:

- [Pattern matching][] and the different [types of patterns][].
- [Switch expressions][] with support for
  patterns and [exhaustiveness checking][].
- [If statements with case clauses][] to support pattern matching.
- [Records][], a new anonymous, immutable, and aggregate type
  that enables [multiple returns][].
- [Class modifiers][] that give libraries more control over exported types.
  As well as a [reference][class-modifier-reference] to
  better understand how they combine.

To help you transition to Dart 3's enforcement of [sound null safety][]
and other changes, we also prepared the following updates:

- Created a comprehensive [Dart 3 migration guide][].
- Migrated all documentation and code examples
  to Dart 3, the latest tool releases, and the latest dependencies.

> - Clarified Dart's null-safe type system and 
>   how that always applies in Dart 3.

Reviewer: ```suggestion
- Clarified that type system is now always null-safe in Dart 3.
```

> - Updated and reorganized the [language evolution][] page
>   and its [language versioning][] discussion.

Reviewer: ```suggestion
- Updated and reorganized the [Language evolution][] page
  and its [Language versioning][] discussion.
```

- Removed remnants of Dart 1 and early Dart 2 docs, notes, and resources.

> In addition to all of that Dart 3 goodness and general site improvements,
> we made the following changes:

Reviewer: Hahah "goodness" is a good one!
```suggestion
In addition to the new Dart 3 content and corresponding updates across the site,
we made the following changes:
```
